content/flare/consensus.md

```diff
@@ -287,3 +287,223 @@ Market forces cause quorum slices to overlap and this gives rise to the network-
+
+The basic inter-node network operation in Blockmania consists of each node periodically broadcasting a block to all other nodes.
+The nodes are loosely synchronized, using a [byzantine clock synchronization](https://lamport.azurewebsites.net/pubs/clocks2.pdf) and emit blocks at regular intervals.
+Blocks from each node form a sequence and each refers to the hash of the previous one, forming a hash chain.
+A block contains information about its creator node, its sequence number in the chain, a sequence of entries representing the content of the block.
+Each block emitted by a node is first a _candidate_ block and contains a position defined by its creator node, its sequence number and its content.
+Each entry in the block content is either a transaction received from a client or a reference to the hash of a valid block received from another node.
+The full block is signed using the private key of its creator and broadcast to all other nodes.
+
+Besides broadcasting blocks, nodes listen to requests from other nodes for specific blocks, and respond by sending them the full contents of the blocks requested.
+
+An honest node includes all transactions received from clients as entries into a block it emits.
+Also, all honest nodes receive and checks for validity all blocks they directly or indirectly reference in their created blocks.
+This means that an honest node receives and stores locally a copy of the full DAG record, starting from the genesis vertex until the last block it emits.
+And if an honest node emits a block, all honest nodes will eventually receive and consider valid (after validating) this block as well as all blocks that it references directly or indirectly in its content entries.
+
+A dishonest node may contradict itself and send two distinct blocks for the same position in the chain or not send a block for a position for some time or ever.
+All honest nodes reference all valid blocks in their own blocks, including contradictory ones, and proceed with the next phase of the protocol called "interpretation".
+
+The aim of the interpretation phase is for all honest nodes, despite holding a different subset of the DAG record at any point, to eventually agree on a specific block for any position or alternatively to agree that there is no such block, and to produce the next position in the total ordering of blocks.
+Blockmania is created as a simplified variant of PBFT algorithm and all parties need to agree on a single position rather than the sequence and all nodes perform their interpretation process independently.
+
+For reaching a decision for a single position, Blockmania runs an abstract terminating reliable broadcast protocol that is never materialized in real exchanged messages but rather the structure of the block DAG is interpreted by each node.
+At any time, a node can emit a message to propose a block in a "pre-prepare" message to all nodes.
```

Comment:
Who's in charge of naming at Blockmania?! 😄 `Pre-prepare` is terrible :p

Reply:
Yes it is, these are three types of messages in their algorithm steps - pre-prepare, prepare, commit.
To be honest, I was trying hard to avoid mentioning them, but in the end decided to include as brief as possible description of these steps because it seemed reasonable to me to explain at least this